local Unittests failing: timezone dependent.

It seems that the test still depends on local timezone. It should not.
Here is the test logs for python3.6
```
self = <test_utils.TestBarmanEncoderV2 object at 0x10f41c2b0>

    @pytest.mark.skipif(
        sys.version_info < (3, 6), reason="Requires Python 3.6 or higher"
    )
    def test_datetime_object_v3(self):
        dt = datetime(2015, 1, 10, 12, 34, 56)
    
        # Test date without tzinfo. should raise an exception
        with pytest.raises(ValueError):
            json.dumps(dt, cls=barman.utils.BarmanEncoderV2)
        dttz = dt.astimezone()
        tz_str = datetime.now().astimezone().isoformat()[-6:]
        assert re.match(r"^[\+-][0-1][0-9]:[0-6][0-9]$", tz_str)
>       assert json.dumps(
            dttz, cls=barman.utils.BarmanEncoderV2
        ) == '"2015-01-10T12:34:56{}"'.format(tz_str)
E       assert '"2015-01-10T12:34:56+01:00"' == '"2015-01-10T12:34:56+02:00"'
E         - "2015-01-10T12:34:56+02:00"
E         ?                       ^
E         + "2015-01-10T12:34:56+01:00"
E         ?                       ^
```

The same goes with python2.7 test .

--- a/tests/requirements_dev.txt
+++ b/tests/requirements_dev.txt
@@ -7,3 +7,6 @@ pytest
 mock
 pytest-timeout==1.4.2
 py
+# msrest is a transitive dependency of azure-storage-blob and
+# is broken on python 2 for versions >= 0.7.0
+msrest<0.7.0

--- a/tests/requirements_minimal.txt
+++ b/tests/requirements_minimal.txt
@@ -9,3 +9,6 @@ mock
 pytest-timeout==1.4.2
 argcomplete
 python-dateutil==1.5
+# msrest is a transitive dependency of azure-storage-blob and
+# is broken on python 2 for versions >= 0.7.0
+msrest<0.7.0

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -557,21 +557,21 @@ class TestBarmanEncoderV2(object):
     @pytest.mark.skipif(
         sys.version_info < (3, 6), reason="Requires Python 3.6 or higher"
     )
-    def test_datetime_object_v3(self):
+    def test_datetime_object_from_py36(self):
         dt = datetime(2015, 1, 10, 12, 34, 56)
 
         # Test date without tzinfo. should raise an exception
         with pytest.raises(ValueError):
             json.dumps(dt, cls=barman.utils.BarmanEncoderV2)
         dttz = dt.astimezone()
-        tz_str = datetime.now().astimezone().isoformat()[-6:]
+        tz_str = dttz.isoformat()[-6:]
         assert re.match(r"^[\+-][0-1][0-9]:[0-6][0-9]$", tz_str)
         assert json.dumps(
             dttz, cls=barman.utils.BarmanEncoderV2
         ) == '"2015-01-10T12:34:56{}"'.format(tz_str)
 
     @pytest.mark.skipif(sys.version_info >= (3, 6), reason="Requires Python 2 or 3.5")
-    def test_datetime_object_v2(self):
+    def test_datetime_object_before_py36(self):
         dt = datetime(2015, 1, 10, 12, 34, 56)
 
         # Test date without tzinfo. should raise an exception
@@ -579,7 +579,7 @@ class TestBarmanEncoderV2(object):
             json.dumps(dt, cls=barman.utils.BarmanEncoderV2)
 
         dttz = dt.replace(tzinfo=tz.tzlocal())
-        tz_str = datetime.now(tz=tz.tzlocal()).isoformat()[-6:]
+        tz_str = dttz.isoformat()[-6:]
         assert re.match(r"^[\+-][0-1][0-9]:[0-6][0-9]$", tz_str)
         assert json.dumps(
             dttz, cls=barman.utils.BarmanEncoderV2

